js/app/models/fcEventModel.js

```diff
@@ -135,17 +136,27 @@ app.factory('FcEvent', function(SimpleEvent) {
 		};
 
 		/**
-		 * lock fc event for editing
+		 * Show undo notification for deletion
+		 * @returns {Promise}
 		 */
-		iface.lock = function() {
-			context.lock = true;
-		};
-
-		/**
-		 * unlock fc event
-		 */
-		iface.unlock = function() {
-			context.lock = false;
+		iface.delete = function() {
+			return new Promise(function(resolve, reject) {
+				context.isDeleted = true;
+
+				const timeout = $timeout(function() {
+					if (context.isDeleted) {
+						resolve();
+					}
+				}, 7500);
```

Comment:
7.5 seconds is too high IMHO. The user has got plenty of time to switch to another app, resulting in the deletion action not executing (correct ?).

I don't know if there's a way to execute queue actions when we're leaving the app. Leaving the browser could be handled though, right ?